app/controllers/katello/concerns/authorization/api/v2/content_views_controller.rb

```diff
@@ -118,14 +118,17 @@ def authorize_remove_environments(view, options)
       # and also ensure that the environments have the remove permission
       return deny_access unless KTEnvironment.promotable.where(:id => env_ids).count == env_ids.size && view.promotable_or_removable?
 
-      if Katello::System.where(:content_view_id => view, :environment_id => env_ids).count > 0
+      total_count = Katello::Host::ContentFacet.where(:content_view_id => view, :lifecycle_environment_id => env_ids).count
+      if total_count > 0
         unless options[:system_content_view_id] && options[:system_environment_id]
```

Comment:
I'd be OK with a separate PR that adds a new parameter and deprecates this parameter.